controllers/extensions/addon_controller_stages.go

```diff
@@ -1227,6 +1236,15 @@ func validateVersion(annotations, kbVersion string) (bool, error) {
 	return validate, nil
 }
 
+func checkAddonSpec(addon *extensionsv1alpha1.Addon) error {
+	if addon.Spec.Type == "Helm" {
+		if addon.Spec.Helm == nil || addon.Spec.Helm.ChartLocationURL == "" {
```

Comment:
I think check addon.Spec.Helm == nil is enough, since addon.Spec.Helm.ChartLocationURL is a required field, in other words, if addon.Spec.Helm != nil then  addon.Spec.Helm.ChartLocationURL cannot be empty.

Reply:
I do have doubts about this point too. Actually, XValidation ensures that when type = Helm, Spec.Helm cannot be nil. However, without XValidation, I can indeed construct a scenario where helm is not nil but ChartLocationURL is empty. Nevertheless, such a scenario might not occur in practical situations.

Reply:
What does you mean by "can indeed construct a scenario where helm is not nil but ChartLocationURL is empty." and "Nevertheless, such a scenario might not occur in practical situations."